Sources/MongoSwift/SDAM.swift

```diff
@@ -231,29 +254,111 @@ public struct TopologyDescription {
         }
     }
 
+    /// An internal function used to determine if there are suitable servers to read from in a replica set.
+    internal func hasSuitableReadServer(_ readPref: ReadPreference) -> Bool {
+        var suitableModes: [ServerType]
+
+        switch readPref.mode {
+        case ReadPreference.Mode.primary:
+            suitableModes = [ServerType.rsPrimary]
+        case ReadPreference.Mode.secondary:
+            suitableModes = [ServerType.rsSecondary]
+        default:
+            suitableModes = [ServerType.rsSecondary, ServerType.rsPrimary]
+        }
+
+        var candidates: [ServerDescription] = servers.filter { suitableModes.contains($0.type) }
+
+        // If read preference specifies a max staleness, first filter out too stale servers
+        if let maxStalenessSeconds = readPref.maxStalenessSeconds {
+            candidates = candidates.filter { (candidate: ServerDescription) in
```

Comment:
can probably omit type annotation for candidate here

Reply:
done